### ISSUE TITLE
Home Brew Support for MacOS Builds

### ISSUE THREAD

**Issue by nikhilbadyal** [AChep/keyguard-app#118]
How about [adding](https://docs.brew.sh/Adding-Software-to-Homebrew#casks) HomeBrew Support ?

**Comment by nikhilbadyal** [comment]
[Raised PR ](https://github.com/Homebrew/homebrew-cask/pull/187250)

**Comment by AChep** [comment]
@nikhilbadyal do you know if the cask is automatically updated later on? Looking at the commits there it seems like there's a bot or something that checks the latest version periodically and updates the casks. Let me know if it needs something from my side for the auto-update.

**Comment by nikhilbadyal** [comment]
Yes. It auto updates whenever you make a release. 
So, nothing needed from your end.

**Comment by nikhilbadyal** [comment]
It's live [now](https://formulae.brew.sh/cask/keyguard). You may close the ticket.

**Comment by nikhilbadyal** [comment]
Hey @AChep , 

Can you maintain a pattern in the release name. I see you have started supplying arm build. But the nomenclature is not consistent. 
Can we have **intel** or **x86_64** in the other build. Just to have a pattern in the URL. 

You can check the mac release at this repo for an [example](https://github.com/ankitects/anki/releases/tag/24.10beta2). It contains 2 build for arm and intel and both are prefixed. 


Why ?

This is needed to have a pattern in the URL to automatically download the relevant file based on the system. This will be easily be maintainable if both release follow common nomenclature. As an example you can check this [out](https://github.com/Homebrew/homebrew-cask/blob/9008f1582575759d4f5cbcd35c7ed667ddd3a930/Casks/a/anki.rb#L9). In this sample cask you can see the type of build is dynamically injected based on the type of system user have.

**Comment by AChep** [comment]
Right, the idea was actually to keep intel builds without a suffix to not break the current cask script. I wasn't sure if that is possible to provide different binaries for different architectures.

I guess what currently works is 
```
arch arm: "-arm64", intel: ""
url "https://github.com/AChep/keyguard-app/releases/download/r#{version.csv.second}/Keyguard-#{version.csv.first}-#{arch}.dmg"
```

but since you are here to quickly fix it, i'm all for going for -intel -mac prefixes

**Comment by AChep** [comment]
@nikhilbadyal now all future releases will have a consistent `-apple` / `-intel` suffixes. I've also renamed the artifacts of the latest release to match it.

**Comment by nikhilbadyal** [comment]
Sure. You can make changes for **-intel** and **-mac**. I'll update the cask ASAP. 

Another Request - 

Not sure how much possible it is but can we maintain the tag name same as version name ? 
A minor convenience with the current setup is the generation of download URL.
 
To bump the version i need to [supply 2 values](https://github.com/Homebrew/homebrew-cask/blob/9008f1582575759d4f5cbcd35c7ed667ddd3a930/Casks/k/keyguard.rb#L2) **Release Version & Tag Name**


```
version "1.6.1,20241004"
```

And the URL gets created like this 

```
url "https://github.com/AChep/keyguard-app/releases/download/r#{version.csv.second}/Keyguard-#{version.csv.first}-#{arc}.dmg"
```

If the Tag is same/similar then i can bump the release just with 1 values. The new cask code will be 


```
version "1.6.1"
```

And the URL gets created like this 

```
url "https://github.com/AChep/keyguard-app/releases/download/r#{version}/Keyguard-#{version}-#{arc}.dmg"
```

Check out this sample cask [here](https://github.com/Homebrew/homebrew-cask/blob/9008f1582575759d4f5cbcd35c7ed667ddd3a930/Casks/a/anki.rb#L9).

**Comment by AChep** [comment]
The tag one is quite a bit more complex. I do plan to improve in this area, as I'm also not a big fan of the current double tag situation. However, I do not currently have enough time to thoroughly plan and implement the requires changes. I will notify you in advance when I'll be making changes to release tags / names tho :) Thanks!

**Comment by AChep** [comment]
I see that you have to do some work for the new release to appear in brew (https://github.com/Homebrew/homebrew-cask/pull/187458). I could just create a PR like that as a part of the release job, if that helps.
@nikhilbadyal

**Comment by nikhilbadyal** [comment]
> However, I do not currently have enough time to thoroughly plan and implement the requires changes.

No worries. It's not something which is causing any issues. So you can take your time.

**Comment by nikhilbadyal** [comment]
Raised the PR for [intel & mac ](https://github.com/Homebrew/homebrew-cask/pull/187689) builds. Fixes https://github.com/AChep/keyguard-app/issues/606 as well.

**Comment by nikhilbadyal** [comment]
hey @AChep , You can use [this](https://github.com/dawidd6/action-homebrew-bump-formula?tab=readme-ov-file).

You can find example [here](https://github.com/asciidoc-py/asciidoc-py/blob/21e33efe96ba9a51d99d1150691dae750afd6ed1/.github/workflows/release.yml#L63). Many more examples [here](https://github.com/dawidd6/action-homebrew-bump-formula?tab=readme-ov-file#examples).
